extensions/amp-access/0.1/amp-access.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assertHttpsUrl} from '../../../src/url';
+import {getService} from '../../../src/service';
+import {installStyles} from '../../../src/styles';
+import {isExperimentOn} from '../../../src/experiments';
+import {log} from '../../../src/log';
+
+
+/*eslint no-unused-vars: 0*/
+/**
+ * The configuration properties are:
+ * - authorization: The URL of the Authorization endpoint.
+ * - pingback: The URL of the Pingback endpoint.
+ * - login: The URL of the Login Page.
+ *
+ * @typedef {{
+ *   authorization: string,
+ *   pingback: string,
+ *   login: string
+ * }}
+ */
+let AccessConfig;
+
+/** @const {!Function} */
+const assert = AMP.assert;
+
+/** @const */
+const EXPERIMENT = 'amp-access';
+
+/** @const */
+const TAG = 'AmpAccess';
+
+
+/**
+ * AccessService implements the complete lifecycle of the AMP Access system.
+ */
+export class AccessService {
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    /** @const {!Window} */
+    this.win = win;
+    installStyles(this.win.document, $CSS$, () => {});
+
+    /** @const @private {boolean} */
+    this.isExperimentOn_ = isExperimentOn(this.win, EXPERIMENT);
+
+    const accessElement = document.getElementById('amp-access');
+
+    /** @const @private {boolean} */
+    this.enabled_ = !!accessElement;
+    if (!this.enabled_) {
+      return;
+    }
+
+    /** @const @private {!Element} */
+    this.accessElement_ = accessElement;
+
+    /** @const {!AccessConfig} */
+    this.config_ = this.buildConfig_();
+  }
+
+  /**
+   * @return {!AccessConfig}
+   * @private
+   */
+  buildConfig_() {
+    let config;
+    try {
+      config = JSON.parse(this.accessElement_.textContent);
+    } catch (e) {
+      throw new Error('Failed to parse "amp-access" JSON: ' + e);
+    }
+    return {
+      authorization: assertHttpsUrl(assert(config['authorization'],
```

Comment:
At this time we don't have an element context yet. Right now, `accessElement` is a `script` element  with `application/json` type. Thus adding error styling to it won't work.